spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -799,6 +799,9 @@ def write_data_spec(
 
         self._weight_scales[placement] = weight_scales
 
+    def clear_connection_cache(self):
+        self._retrieved_blocks = dict()
```

Comment:
Just a thought - if we know what each of the edge synapse types are, we could avoid re-reading the synaptic data when the connection doesn't use plasticity.  The "clear" here would then iterate through the retrieved blocks and only reset those where the connection isn't plastic.  The synapse info is known at the time of retrieval, so this information is available to be stored in the retrieved block dict.